config/config.go

```diff
@@ -68,6 +68,99 @@ func NewCloudConfig(contents string) (*CloudConfig, error) {
 	return &cfg, err
 }
 
+func (co *CoreOS) Merge(mergeWith CoreOS) {
```

Comment:
The intent of the config package is to be the minimum amount of code to represent and validate a config. It is consumed by a number of other projects. Can you move the merging-related code out of this package?